cmd/zrepl/mgmt_conn.c

```diff
@@ -538,6 +560,115 @@ uzfs_zvol_rebuild_status(uzfs_mgmt_conn_t *conn, zvol_io_hdr_t *hdrp,
 	return (reply_data(conn, &hdr, &status_ack, sizeof (status_ack)));
 }
 
+/*
+ * Iterate through all finished async tasks and send replies to clients.
+ */
+static int
+finish_async_tasks(void)
+{
+	uzfs_mgmt_conn_t *conn;
+	async_arg_t *async_task;
+	int rc;
+
+	mutex_enter(&async_tasks_mtx);
+	while ((async_task = SLIST_FIRST(&async_tasks)) != NULL) {
+		SLIST_REMOVE_HEAD(&async_tasks, task_next);
+		/* connection could have been removed in the meantime */
+		mutex_enter(&conn_list_mtx);
+		SLIST_FOREACH(conn, &uzfs_mgmt_conns, conn_next) {
+			if (conn == async_task->ctx)
+				break;
+		}
+		mutex_exit(&conn_list_mtx);
+		if (conn != NULL) {
+			rc = reply_nodata(conn, async_task->status,
+			    async_task->hdr.opcode, async_task->hdr.io_seq);
+		}
+		kmem_free(async_task, sizeof (*async_task));
+		if (rc != 0)
+			return (rc);
+	}
+	mutex_exit(&async_tasks_mtx);
+	return (0);
+}
+
+/*
+ * Perform the command (in async context).
+ *
+ * Currently we have only snapshot commands which are async. We might need to
+ * make the code & structures more generic if we add more commands.
+ */
+static void
+uzfs_zvol_do_async_command(void *arg)
```

Comment:
Suggestion ...._do_async_ can be replaced with _execute_async_.

Reply:
will do. thanks